### DISCUSSION TITLE
GGUF vs bin?

### DISCUSSION THREAD

**Discussion by VadimBoev** [ggml-org/llama.cpp#6068] [Q&A] [answered]
Hi everybody. I'm sorry if I asked an incorrect question, I'm not good at this.  
I am interested in maximum performance. Which is better than GGUF or bin?

Also, I saw a topic in which ggerganov advised to disable map and compile llama.cpp after such changes.  
Does it really work? And how can I set parameters for high performance?  
All I managed to do was launch "./main -m ../openchat-3.5-0106.Q4_K_M.of "Building a website can be done in 10 simple steps:\step 1:" -n 1024 -e"

**Comment by Jeximo** [comment, marked as answer]
> I am interested in maximum performance. Which is better than GGUF or bin

Same performance.

> Also, I saw a topic in which ggerganov advised to disable map and compile llama.cpp after such changes.
Does it really work?

`--no-mmap` in the command line may help performance, depending on your system RAM [--no-mmap](https://github.com/ggerganov/llama.cpp/blob/4e9a7f7f7fb6acbddd1462909c8d696e38edbfcc/examples/main/README.md?plain=1#L282)

> And how can I set parameters for high performance?

Add `--threads N`, [-t N, --threads N](https://github.com/ggerganov/llama.cpp/tree/4e9a7f7f7fb6acbddd1462909c8d696e38edbfcc/examples/main#performance-tuning-and-memory-options) 

If you build including GPU, then also add `-ngl N` [-ngl N, --n-gpu-layers N](https://github.com/ggerganov/llama.cpp/tree/4e9a7f7f7fb6acbddd1462909c8d696e38edbfcc/examples/main#additional-options)

**Comment by VadimBoev** [comment]
thank you!
